homeassistant/components/hyperion/light.py

```diff
@@ -26,41 +26,24 @@
 CONF_HDMI_PRIORITY = "hdmi_priority"
 CONF_EFFECT_LIST = "effect_list"
 
+# As we want to preserve brightness control for effects (e.g. to reduce the
+# brightness for V4L), we need to persist the effect that is in flight, so
+# subsequent calls to turn_on will know the keep the effect enabled.
+# Unfortunately the Home Assistant UI does not easily expose a way to remove a
+# selected effect (there is no 'No Effect' option by default). Instead, we
+# create a new fake effect ("Solid") that is always selected by default for
+# showing a solid color. This is the same method used by WLED.
+KEY_EFFECT_SOLID = "Solid"
+
 DEFAULT_COLOR = [255, 255, 255]
+DEFAULT_BRIGHTNESS = 255
+DEFAULT_EFFECT = KEY_EFFECT_SOLID
 DEFAULT_NAME = "Hyperion"
+DEFAULT_ORIGIN = "Home Assistant"
 DEFAULT_PORT = 19444
 DEFAULT_PRIORITY = 128
 DEFAULT_HDMI_PRIORITY = 880
-DEFAULT_EFFECT_LIST = [
-    "HDMI",
-    "Cinema brighten lights",
-    "Cinema dim lights",
-    "Knight rider",
-    "Blue mood blobs",
-    "Cold mood blobs",
-    "Full color mood blobs",
-    "Green mood blobs",
-    "Red mood blobs",
-    "Warm mood blobs",
-    "Police Lights Single",
-    "Police Lights Solid",
-    "Rainbow mood",
-    "Rainbow swirl fast",
-    "Rainbow swirl",
-    "Random",
-    "Running dots",
-    "System Shutdown",
-    "Snake",
-    "Sparks Color",
-    "Sparks",
-    "Strobe blue",
-    "Strobe Raspbmc",
-    "Strobe white",
-    "Color traces",
-    "UDP multicast listener",
-    "UDP listener",
-    "X-Mas",
-]
+DEFAULT_EFFECT_LIST = []
```

Comment:
We can deprecate config options using `cv.deprecated`.

Reply:
A prior reviewer for an earlier version of this PR attempt (which didn't have the full client refactor you see here) asked to leave the YAML untouched due to ADR-0007 and ADR-0010 decisions. My original PR used cv.deprecated as you suggest here.

My intent is, once this is submitted, to rip out the YAML entirely and replace it with a config_flow (only).

I am happy to go either way. Your advice welcome on which direction to take (and indeed whether ripping the YAML out entirely soon after is acceptable in this case).

Reply:
If we are not using the yaml option in the code we need to notify the user of this. We won't invalidate the option now, just deprecate it. The schema should stay but we should add `cv.deprecated` to the options that the code is ignoring.

Reply:
Got it -- thank you!